### PR TITLE
Add storage of entity proxy along with the classification operation

### LINKED ISSUE #5836: AnchorClassification created on get
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

During a getEntityDetails operation, the OpenMetadataAPIGenericHandler validates the Anchor classification which leads to creating it. Since the entity that was searched for is actually coming from external repo, it is not found in the local repo and therefore the following exception, compromising vertical lineage build up
Also tested by commenting out the call to validateAnchorEntity from OpenMetadataAPIGenericHandler:9132 and there were no exceptions in logs, all went smoothly

See below the stacktrace

### Expected Behavior

No exceptions, vertical lineage is not compromised

### Steps To Reproduce

- update a GlossaryTerm in external repo with a new SemanticAssignment
- wait for the modification to be picked up by Egeria

### Environment

```markdown
- Egeria:
- OS:
- Java:
- Browser (for UI issues):
- Additional connectors and integration:
```

### Any Further Information?

Stacktrace

Wed Oct 20 18:41:27 EEST 2021 omas Exception OMAG-REPOSITORY-HANDLER-0003 An unexpected error org.odpi.openmetadata.repositoryservices.ffdc.exception.EntityNotKnownException was returned to classifyEntity by the metadata server during getEntityDetails request for open metadata access service Asset Lineage OMAS on server omas; message was OMRS-GRAPH-REPOSITORY-400-004 The attempt to retrieve an entity with GUID term@2103bfbe-9851-47c1-96fe-a26c972758bc:6662c0f2.e1b1ec6c.66k7ou9o9.rqof6p4.55hq6r.arhfdeut2kd8jnj6slkcn failed in getEntityDetailFromStore method of class org.odpi.openmetadata.adapters.repositoryservices.graphrepository.repositoryconnector.GraphOMRSMetadataStore to open metadata repository omas
Wed Oct 20 18:41:27 EEST 2021 omas Exception OMAG-REPOSITORY-HANDLER-0003 Supplementary information: log record id d0340583-16ce-4190-9bef-2827bb794e99 org.odpi.openmetadata.repositoryservices.ffdc.exception.EntityNotKnownException returned message of OMRS-GRAPH-REPOSITORY-400-004 The attempt to retrieve an entity with GUID term@2103bfbe-9851-47c1-96fe-a26c972758bc:6662c0f2.e1b1ec6c.66k7ou9o9.rqof6p4.55hq6r.arhfdeut2kd8jnj6slkcn failed in getEntityDetailFromStore method of class org.odpi.openmetadata.adapters.repositoryservices.graphrepository.repositoryconnector.GraphOMRSMetadataStore to open metadata repository omas and stacktrace of 
OCFCheckedExceptionBase{reportedHTTPCode=400, reportingClassName='org.odpi.openmetadata.adapters.repositoryservices.graphrepository.repositoryconnector.GraphOMRSMetadataStore', reportingActionDescription='getEntityDetailFromStore', reportedErrorMessage='OMRS-GRAPH-REPOSITORY-400-004 The attempt to retrieve an entity with GUID term@2103bfbe-9851-47c1-96fe-a26c972758bc:6662c0f2.e1b1ec6c.66k7ou9o9.rqof6p4.55hq6r.arhfdeut2kd8jnj6slkcn failed in getEntityDetailFromStore method of class org.odpi.openmetadata.adapters.repositoryservices.graphrepository.repositoryconnector.GraphOMRSMetadataStore to open metadata repository omas', reportedErrorMessageId='OMRS-GRAPH-REPOSITORY-400-004', reportedErrorMessageParameters=[term@2103bfbe-9851-47c1-96fe-a26c972758bc:6662c0f2.e1b1ec6c.66k7ou9o9.rqof6p4.55hq6r.arhfdeut2kd8jnj6slkcn, getEntityDetailFromStore, org.odpi.openmetadata.adapters.repositoryservices.graphrepository.repositoryconnector.GraphOMRSMetadataStore, omas], reportedSystemAction='The system was unable to perform the entity retrieval.', reportedUserAction='Correct the caller's code and retry the entity retrieval request.', reportedCaughtException=null, reportedCaughtExceptionClassName='null', relatedProperties=null}
	at org.odpi.openmetadata.adapters.repositoryservices.graphrepository.repositoryconnector.GraphOMRSMetadataStore.getEntityDetailFromStore(GraphOMRSMetadataStore.java:585)
	at org.odpi.openmetadata.adapters.repositoryservices.graphrepository.repositoryconnector.GraphOMRSMetadataCollection.getEntityDetail(GraphOMRSMetadataCollection.java:415)
	at org.odpi.openmetadata.repositoryservices.localrepository.repositoryconnector.LocalOMRSMetadataCollection.validateEntityIsNotProxy(LocalOMRSMetadataCollection.java:3342)
	at org.odpi.openmetadata.repositoryservices.localrepository.repositoryconnector.LocalOMRSMetadataCollection.validateEntityCanBeClassified(LocalOMRSMetadataCollection.java:3456)
	at org.odpi.openmetadata.repositoryservices.localrepository.repositoryconnector.LocalOMRSMetadataCollection.classifyEntity(LocalOMRSMetadataCollection.java:4457)
	at org.odpi.openmetadata.repositoryservices.enterprise.repositoryconnector.executors.ClassifyEntityExecutor.issueRequestToRepository(ClassifyEntityExecutor.java:100)
	at org.odpi.openmetadata.repositoryservices.enterprise.repositoryconnector.control.SequentialFederationControl.executeCommand(SequentialFederationControl.java:58)
	at org.odpi.openmetadata.repositoryservices.enterprise.repositoryconnector.EnterpriseOMRSMetadataCollection.classifyEntity(EnterpriseOMRSMetadataCollection.java:3276)
	at org.odpi.openmetadata.commonservices.repositoryhandler.RepositoryHandler.classifyEntity(RepositoryHandler.java:1879)
	at org.odpi.openmetadata.commonservices.generichandlers.OpenMetadataAPIGenericHandler.maintainAnchorGUIDInClassification(OpenMetadataAPIGenericHandler.java:2384)
	at org.odpi.openmetadata.commonservices.generichandlers.OpenMetadataAPIGenericHandler.validateAnchorEntity(OpenMetadataAPIGenericHandler.java:3773)
	at org.odpi.openmetadata.commonservices.generichandlers.OpenMetadataAPIGenericHandler.getEntityFromRepository(OpenMetadataAPIGenericHandler.java:9132)
	at org.odpi.openmetadata.commonservices.generichandlers.OpenMetadataAPIGenericHandler.getEntityFromRepository(OpenMetadataAPIGenericHandler.java:9069)
	at org.odpi.openmetadata.accessservices.assetlineage.handlers.HandlerHelper.getEntityDetails(HandlerHelper.java:209)
	at org.odpi.openmetadata.accessservices.assetlineage.handlers.HandlerHelper.buildContextForRelationships(HandlerHelper.java:354)
	at org.odpi.openmetadata.accessservices.assetlineage.handlers.GlossaryContextHandler.buildGlossaryTermContext(GlossaryContextHandler.java:98)
	at org.odpi.openmetadata.accessservices.assetlineage.outtopic.AssetLineagePublisher.publishGlossaryContext(AssetLineagePublisher.java:137)
	at org.odpi.openmetadata.accessservices.assetlineage.server.AssetLineageRestServices.publishContext(AssetLineageRestServices.java:243)
	at org.odpi.openmetadata.accessservices.assetlineage.server.AssetLineageRestServices.publishEntityContext(AssetLineageRestServices.java:213)
	at org.odpi.openmetadata.accessservices.assetlineage.server.AssetLineageRestServices.lambda$buildAndPublishLineageContext$2(AssetLineageRestServices.java:152)
	at org.odpi.openmetadata.accessservices.assetlineage.server.AssetLineageRestServices.lambda$buildAndPublishLineageContext$3(AssetLineageRestServices.java:152)

### PR DESCRIPTION
**Pull request by lcpopa**
Signed-off-by: Liviu Popa <liviu-constantin.popa@ing.com>

# Description

Use case:

- ols requests latest changes from al-omas which interogates igc via igc-proxy
- igc-proxy returns latest changes to al-omas which builds the context for returned entities
- classifying fails on returned entities because they do not exist in local repo
- added ability to save their proxies when classifying them

PR does not contain event processing changes for classify actions.

Fixes #5836 

# How Has This Been Tested?

Automatic and manual on described use case. Succesfuly executed jupyter lab "Building a Data Catalog"

# Any additional notes for reviewers?

### CODE REVIEW

File: open-metadata-implementation/repository-services/repository-services-implementation/src/main/java/org/odpi/openmetadata/repositoryservices/localrepository/repositorycontentmanager/OMRSRepositoryContentHelper.java [hunk 1]
```diff
@@ -1606,6 +1643,44 @@ public EntityDetail updateClassificationInEntity(String         sourceName,
...
+
+            return this.addClassificationToEntity(sourceName, entity, updatedClassification, methodName);
+        }
+        else
+        {
+            final String thisMethodName = "updateClassificationInEntity";
+
+            throw new OMRSLogicErrorException(OMRSErrorCode.NULL_CLASSIFICATION_CREATED.getMessageDefinition(sourceName,
+                    thisMethodName,
+                    methodName),
```
Comment (mandy-chessell) [inline open-metadata-implementation/repository-services/repository-services-implementation/src/main/java/org/odpi/openmetadata/repositoryservices/localrepository/repositorycontentmanager/OMRSRepositoryContentHelper.java:1677]: Indent parameters

File: open-metadata-implementation/repository-services/repository-services-implementation/src/main/java/org/odpi/openmetadata/repositoryservices/localrepository/repositorycontentmanager/OMRSRepositoryContentHelper.java [hunk 2]
```diff
@@ -1681,6 +1756,80 @@ public EntityDetail deleteClassificationFromEntity(String       sourceName,
...
+                }
+            }
+
+            Classification oldClassification = entityClassificationsMap.remove(oldClassificationName);
+
+            if (oldClassification == null)
+            {
+                throw new ClassificationErrorException(OMRSErrorCode.ENTITY_NOT_CLASSIFIED.getMessageDefinition(methodName,
+                        sourceName,
+                        oldClassificationName,
```
Comment (mandy-chessell) [inline open-metadata-implementation/repository-services/repository-services-implementation/src/main/java/org/odpi/openmetadata/repositoryservices/localrepository/repositorycontentmanager/OMRSRepositoryContentHelper.java:1803]: Indent parameters

File: open-metadata-implementation/repository-services/repository-services-implementation/src/main/java/org/odpi/openmetadata/repositoryservices/localrepository/repositorycontentmanager/OMRSRepositoryContentHelper.java [hunk 3]
```diff
@@ -1681,6 +1756,80 @@ public EntityDetail deleteClassificationFromEntity(String       sourceName,
...
+
+            return updatedEntity;
+        }
+        else
+        {
+            final String thisMethodName = "deleteClassificationFromEntity";
+
+            throw new OMRSLogicErrorException(OMRSErrorCode.NULL_CLASSIFICATION_NAME.getMessageDefinition(sourceName,
+                    thisMethodName,
+                    methodName),
```
Comment (mandy-chessell) [inline open-metadata-implementation/repository-services/repository-services-implementation/src/main/java/org/odpi/openmetadata/repositoryservices/localrepository/repositorycontentmanager/OMRSRepositoryContentHelper.java:1828]: Indent parameters

**Review by mandy-chessell — DISMISSED** [review]
This is great - comments only on updating the code formatting to match the rest of OMRS.

**Comment by lcpopa** [comment]
Implemented requested changes

**Review by marius-patrascu — APPROVED** [review]

**Review by bogdan-sava — APPROVED** [review]

**Pull request merged**